analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicCheck.cs

```diff
@@ -28,7 +28,7 @@ public class SymbolicCheck
     /// <summary>
     /// Stop processing this branch of the exploded graph. There will be no follow up states.
     /// </summary>
-    protected static readonly ProgramState[] EmptyStates =  Array.Empty<ProgramState>();
+    protected static readonly ProgramStates EmptyStates = new();
```

Comment:
This should be removed. Accessing the field has value semantics and a copy is created (we see a [`ldsfld`](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.emit.opcodes.ldsfld?view=net-8.0) (copies the value for valueue types) instead of a [`ldsflda`](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.emit.opcodes.ldsflda?view=net-8.0) (creates a reference to the field) in the [generated IL](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGED2AbbEDGMAlpgHYAyRMECU2yANDCEdgwCYgDUAPgAIAmAIwBYAFB8AzAAJB09NIDe46atlCAbNKRR2ZbAE9pAZWkA5OLgAKmIqRgBeaaQgB3ABQAGBtM8BKAG4VNWDVKVkAFmkAWXc/aVClRLVpADcoBGcAB2knC2tbeyCxFJTSLIA6ABVMYxgEOwBzOOKUgF9xDrFxHT1SQ2lkerhCE3FlErVe/SM7GGkAQQCQydVp/tn7aQAhVrVw43c56SgfY+A/BwA+aXcFn234p3dTi+KulKA===)).

We should just `new()` as we do in all the other places as well.